src/compile/data/dataflow.ts

```diff
@@ -0,0 +1,133 @@
+import { isFunction } from 'util';
+import { StringSet } from '../../util';
+/**
+ * A node in the dataflow graph.
+ */
+export class DataFlowNode {
+  private _children: DataFlowNode[];
+
+  private _parent: DataFlowNode;
+
+  constructor(public readonly debugName?: string) {
+    this._children = [];
```

Comment:
Ts requires the super() call so you cannot forget it. Your code won't compile without it.